frontend/wallet/src/render/views/settings/DelegationSettings.re

```diff
@@ -71,20 +88,54 @@ let make = (~publicKey) => {
   let goBack = () =>
     ReasonReact.Router.push("/settings/" ++ PublicKey.uriEncode(publicKey));
 
-  // Mutation variables and handlers
-  let variables =
-    ChangeDelegation.make(
-      ~from=Apollo.Encoders.publicKey(publicKey),
-      ~to_=Apollo.Encoders.publicKey(PublicKey.ofStringExn(state.delegate)),
-      ~fee=
-        Apollo.Encoders.currency(
-          switch (state.fee) {
-          | DefaultAmount => defaultFee
-          | Custom(amount) => amount
-          },
-        ),
-      (),
-    )##variables;
+  let renderError = () => {
+    switch (state.error) {
+    | Some(EmptyDelegateKey) =>
+      <>
+        <Alert kind=`Danger defaultMessage="Please enter a public key" />
+        <Spacer height=2. />
+      </>
+    | Some(InvalidTransactionFee) =>
+      <>
+        <Alert
+          kind=`Danger
+          defaultMessage={"The minimum transaction fee is " ++ minimumFee}
+        />
+        <Spacer height=2. />
+      </>
+    | Some(InvalidGraphQLResponse(error)) =>
+      <> <Alert kind=`Danger defaultMessage=error /> <Spacer height=2. /> </>
+    | None => React.null
+    };
+  };
+
+  let onDelegateClick = (mutate: ChangeDelegationMutation.apolloMutation) => {
+    switch (state.delegate, state.fee) {
+    | ("", _) =>
+      changeState(prev => {...prev, error: Some(EmptyDelegateKey)})
+    | (_, Custom(fee)) =>
+      switch (fee) {
+      | "" =>
+        changeState(prev => {...prev, error: Some(InvalidTransactionFee)})
+      | fee when float_of_string(fee) < float_of_string(minimumFee) =>
+        changeState(prev => {...prev, error: Some(InvalidTransactionFee)})
+      | _ =>
+        mutate(
+          ~variables,
+          ~refetchQueries=[|"getAccountInfo", "queryDelegation"|],
+          (),
+        )
+        |> ignore
+      }
+    | _ =>
+      mutate(
+        ~variables,
+        ~refetchQueries=[|"getAccountInfo", "queryDelegation"|],
+        (),
+      )
+      |> ignore
```

Comment:
I tried to refactor this code duplication but I had problems with switch branches. If there are any thoughts, I'd love thoughts!